chalice/deploy/newdeployer.py

```diff
@@ -517,14 +565,82 @@ def _do_jpsearch(self, instruction):
         result = jmespath.search(instruction.expression, v)
         self.stack.append(result)
 
+    def _do_builtinfunction(self, instruction):
+        # type: (models.BuiltinFunction) -> None
+        # Split this out to a separate class of built in functions
+        # once we add more functions.
+        if instruction.function_name == 'parse_arn':
```

Comment:
Are we hoping to keep this to one builtin? Is there a reason that we chose this way rather than just making a parse arn op code other than room for growth?

Reply:
Primarily it just seemed too specific.  I'd like the opcodes somewhat generic, and `parse_arn` just seemed too specialized.